src/snyk/common/views/webviewProvider.ts

```diff
@@ -25,9 +25,12 @@ export abstract class WebviewProvider<ViewModel> implements IWebViewProvider<Vie
   restorePanel(panel: vscode.WebviewPanel): void {
     if (this.panel) this.panel.dispose();
     this.panel = panel;
+    this.registerListeners();
+    this.panel.webview.html = this.getHtmlForWebview(this.panel.webview);
```

Comment:
There's a need to call some custom logic in [codeSuggestionWebviewProvider.ts](https://github.com/snyk/vscode-extension/pull/248/files#diff-3f69198f2ef12d4c2a39fa3d622621d106fe54db326e22a7d8869a319845760b). It is setting `this.suggestion` on the provider, which doesn't get set from here.

We need it for correct functioning of `checkCurrentSuggestion()` that checks if suggestion panel contains an existing issue from the run analysis.

Is there a way still to reuse `showPanel()` to keep custom provider logic in place?

Reply:
Hmm, I see.. We might want to split showPanel and updatePanel or something of the sort.
I will have a closer look and make a suggestion.

Reply:
You're probably right. You can ditch `checkCurrentSuggestion()` function altogether then. OSS provider doesn't have this check, thus we're consistent.

Reply:
So, the implementation in the PR is consistent. We don't set `this.suggestion` and the check [here](https://github.com/snyk/vscode-extension/blob/main/src/snyk/snykCode/views/suggestion/codeSuggestionWebviewProvider.ts#L70) will just skip it.

Reply:
Yep, let's get this out. 🚀